Dodanie testów automatycznych

MateuszPazio opened the pull request.

MateuszPazio committed:
Dodanie szkieletu testów #36

MateuszPazio committed:
test automatyczny v1

MateuszPazio committed:
Poprawa błędów edycyjnych #36

MateuszPazio committed:
Dostosowanie testów do działania z danymi testowymi #36

wojtek19022 commented on constants.py:
```diff
@@ -15,3 +15,5 @@
     "n": "Nieruchomości"
 }
+
+CSV_URL = "https://downloads.envirosolutions.pl/sample-data.csv"
```
To też jest dodane do README.md? Finalnie z tego poziomu użytkownik powinien móc sobie pobrać dane. Dlatego pytam

wojtek19022 commented on test/test_geokodowanie.py:
```diff
@@ -0,0 +1,176 @@
+
+            return True
+        except Exception as e:
+            self.fail(f"Nie udało się pobrać pliku CSV. Błąd: {e}")
```
Trochę mi się nie podoba ta implementacja. Przede wszystkim lepiej wykorzystać domyślnie wbudowane narzędzie QgsNetworkAccessManager. Mikołaj to już implementował to możesz się go podpytać.

Ponadto jest zbyt szeroki Exception. Faktycznie lepiej go dodać ale na sam koniec. Wcześniej lepiej wyłapać poszczególne typy błędów aby móc jakoś to debugować czy się poprało, a jeżeli nie to z jakim błędem

wojtek19022 commented on test/test_geokodowanie.py:
```diff
@@ -0,0 +1,176 @@
+                miejscowosci.append(miasto)
+                ulice.append(ulica)
+                kody.append(kod)
+                numery.append(numer)
```
Tego fragmentu kodu nie ma we wtyczce geokodowanie? Była tam możliwość wybierania określonych pól w comboboxach tak aby były mapowane przez narzędzie. Możesz Może lepiej to będzie zrobić jako osobne testy, czy w ogóle kolumna jest prawidłowo odnajdywana, ale też raczej powinno byc to przetestowane czy ta kolumna w ogóle jest prawidłowo mapowana przez wtyczkę. Na pewno nigdy nie powinna być pusta (tak jak tam masz wyżej w else)

wojtek19022 commented on test/test_geokodowanie.py:
```diff
@@ -0,0 +1,176 @@
+        print(
+            f"Wynik: Zgeokodowano {total_found}/{oczekiwana_liczba}."
+            f" Błędnych: {total_errors}"
+            )
```
Może lepiej na sam koniec dodać jeszcze jakiś nagłówek z POSUMOWANIEM tak aby dla osoby sprawdzajacej test było to czytelne

wojtek19022 reviewed (changes requested):
do weryfikacji niektóre punkty

MateuszPazio committed:
update testów #36

Przebudowałem poważnie kod testów, ponieważ zainspirowałem się tym co zrobił Mikołaj i jego metoda robienia testów została zakceptowana oraz jest łatwiejsza w implementacji oraz spełnia wskazane komentarze. Test działa robi pdosumowanie i wykrozstuje wbudowane funckje do pobirania csv, Test jest dostosowany do podejścia gdzie mamy nagłówki i mają następująca nuemracje 0:Miasto, 1:Ulica, 2:Numer, 3:Kod. To są elementy które użytkownik wybeira manualnie dlatego zostało wprowadozne takie rozwiązanie, gdzie mamy to zdefiniowane.

wojtek19022 reviewed (approved):
git, merguj
